joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -25,9 +24,26 @@ trait AstCreatorHelper { this: AstCreator =>
   }
 
   private def nodeType(node: Value): ParserNode = fromString(node(ParserKeys.NodeType).str)
-  protected def code(node: Value): String = {
-    // TODO Need to build a utlity which returns code for a node
-    ""
+  protected def code(node: Value): Try[String] = Try {
+
+    val lineNumber    = line(node).get
+    val colNumber     = column(node).get - 1
+    val lineEndNumber = lineEndNo(node).get
+    val colEndNumber  = columnEndNo(node).get - 1
+
+    var nodeCode = ""
```

Comment:
I would discourage the use of mutable variables. `If-else` also returns values. So for this I'd also perhapse encourage a `StringBuilder` since those are faster at building strings. Also, there is an `Iterator` functional style of conditionally iterating which is preferable as it doesn't use mutable variables. See below:
```scala
    if (lineNumber == lineEndNumber) {
       lineNumberMapping(lineNumber).substring(colNumber, colEndNumber)
     } else {
       val sb = new mutable.StringBuilder()
       sb.append(lineNumberMapping(lineNumber).substring(colNumber))
       Iterator
         .from(lineNumber + 1)
         .takeWhile(currentLineNumber => currentLineNumber < lineEndNumber)
         .foreach(currentLineNumber => sb.append(s"\n${lineNumberMapping(currentLineNumber)}"))
       sb.toString()
     }
```

Reply:
Thanks, will make this change